e2e/scripts/actions/persistent_overlay.test

```diff
@@ -0,0 +1,49 @@
+#!/singularity/test
+# persistent overlay tests
+
+source common.func
+
+# check overlay is supported, or skip those tests
+check-overlayfs
+
+# check command presence
+check-command mksquashfs
+check-command dd
+check-command mkfs.ext3
+
+# test staging files/directories
+ext3_overlay_image=`mktmpfile`
+overlaydir=`mktmpdir`
+squashdir=`mktmpdir`
+squash_overlay_img=`mktmpfile`
+echo "bogus" > ${squashdir}/squash_overlay
+
+# create overlay images
+run-command mksquashfs ${squashdir} ${squash_overlay_img} -noappend
+run-command dd if=/dev/zero of=${ext3_overlay_image} bs=1M count=64
+run-command mkfs.ext3 -q -F ${ext3_overlay_image}
+
+# get test image
+image=`get-test-image`
+
+# test directory overlay
+expect-success "TouchWithDirOverlay"        sudo singularity exec -o ${overlaydir} ${image} touch /dir_overlay
```

Comment:
I think we should have a check to see if a user can actually elevate their privileges before running `sudo` and potentially skip if they cannot. This way we can easily test both privileged and non-privileged users.

It would also be good if we could run these as regular users to verify they fail without having to repeat the command without `sudo` attached.

Reply:
This is not required because the whole test suite run sudo at the very beginning (and keep running a sudo true every minute to keep user sudo session in case of long running test), so if user doesn't have right it could not even run tests, like it's done for every `go test` we have with make.